### ISSUE TITLE
Boolean field in dictionary support

### ISSUE THREAD

**Issue by eoger** [mozilla/uniffi-rs#35]
I want to be able to compile the following code:
```
dictionary Point {
  double x;
  double y;
  boolean visited;
};
```

Currently getting errors:
```
error[E0277]: the trait bound `bool: uniffi::support::Lowerable` is not satisfied
  --> /Users/eoger/uniffi-rs/target/debug/build/uniffi-example-geometry-32b57fa3d176b525/out/geometry.uniffi.rs:38:9
   |
38 |         uniffi::support::Lowerable::lower_into(&self.visited, buf);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `uniffi::support::Lowerable` is not implemented for `bool`
   |
  ::: /Users/eoger/uniffi-rs/uniffi/src/support/mod.rs:17:22
   |
17 |     fn lower_into<B: BufMut>(&self, buf: &mut B);
   |                      ------ required by this bound in `uniffi::support::Lowerable::lower_into`

error[E0277]: the trait bound `u8: uniffi::support::Liftable` is not satisfied
  --> /Users/eoger/uniffi-rs/target/debug/build/uniffi-example-geometry-32b57fa3d176b525/out/geometry.uniffi.rs:47:22
   |
47 |             visited: <u8 as uniffi::support::Liftable>::try_lift_from(buf)?,
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `uniffi::support::Liftable` is not implemented for `u8`
   |
  ::: /Users/eoger/uniffi-rs/uniffi/src/support/mod.rs:68:25
   |
68 |     fn try_lift_from<B: Buf>(buf: &mut B) -> Result<Self>;
   |                         --- required by this bound in `uniffi::support::Liftable::try_lift_from`

error[E0308]: try expression alternatives have incompatible types
  --> /Users/eoger/uniffi-rs/target/debug/build/uniffi-example-geometry-32b57fa3d176b525/out/geometry.uniffi.rs:47:22
   |
47 |             visited: <u8 as uniffi::support::Liftable>::try_lift_from(buf)?,
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `u8`
```

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-1555)

**Comment by rfk** [comment]
Similar to https://github.com/mozilla/uniffi-rs/issues/36#issuecomment-665218889, we need to provide builtin implementations of the `Liftable` and `Lowerable` trait for the `bool` type. It would serialize to a single byte, `0x00` for false and `0x01` for true.

Actually it looks like a `ViaFfi` implementation is also missing for the `bool` type, probably just an oversight on my part.

> visited: <u8 as uniffi::support::Liftable>::try_lift_from(buf)?,

Also, I think this should say `bool` instead of `u8` in the type expression here, probably [this line](https://github.com/mozilla/uniffi-rs/blob/3c36c7b70c68a67c3861e4d14efc6c2857213b22/uniffi/src/scaffolding.rs#L32) is left over from a previous code factoring and needs to be updated to output `bool` directly.

**Comment by rfk** [comment]
I think this should be at least partially fixed by #44 above, but there may be stuff left to do in the foreign language bindings.

**Comment by eoger** [comment]
I'll assign myself to add a test to confirm we fixed the behavior
